### PR TITLE
Update protobuf-net to v2.1.0

### PR DESCRIPTION
**Pull request by paralin**
Hi guys,

I really need this update asap because Protobuf.net does not support dotnetstandard until v2.1.0, and I cannot install both protobuf-net v2.1.0 and Steamkit2 in its current version at the same time.

I've bumped the dependencies and would greatly appreciate it if you could spin a release with this update.

Thanks!

### CODE REVIEW

**Comment by paralin** [comment]
Un-bumped xunit, updated samples, should work

File: Samples/3.DebugLog/Sample3_DebugLog.csproj [hunk 1]
```diff
@@ -37,6 +37,9 @@
     <Prefer32Bit>false</Prefer32Bit>
   </PropertyGroup>
   <ItemGroup>
+    <Reference Include="protobuf-net, Version=2.1.0.0, Culture=neutral, PublicKeyToken=257b51d87d2e4d67, processorArchitecture=MSIL">
```
Comment (Netshroud) [inline Samples/3.DebugLog/Sample3_DebugLog.csproj]: Why add a reference to a project that previously did not need it?

File: Samples/1.Logon/Sample1_Logon.csproj [hunk 2]
```diff
@@ -37,6 +37,9 @@
     <Prefer32Bit>false</Prefer32Bit>
   </PropertyGroup>
   <ItemGroup>
+    <Reference Include="protobuf-net, Version=2.1.0.0, Culture=neutral, PublicKeyToken=257b51d87d2e4d67, processorArchitecture=MSIL">
```
Comment (Netshroud) [inline Samples/1.Logon/Sample1_Logon.csproj]: Why add a reference to a project that previously did not need it?

File: Samples/4.Friends/Sample4_Friends.csproj [hunk 3]
```diff
@@ -39,6 +39,9 @@
     <Prefer32Bit>false</Prefer32Bit>
   </PropertyGroup>
   <ItemGroup>
+    <Reference Include="protobuf-net, Version=2.1.0.0, Culture=neutral, PublicKeyToken=257b51d87d2e4d67, processorArchitecture=MSIL">
```
Comment (Netshroud) [inline Samples/4.Friends/Sample4_Friends.csproj]: Same here, etc...

**Comment by Netshroud** [comment]
What's the purpose of this? You've stated that:

>  Protobuf.net does not support dotnetstandard until v2.1.0, and I cannot install both protobuf-net v2.1.0 and Steamkit2 in its current version at the same time.

however, this just leaves me confused.

SteamKit2 does not support .NET Standard either, and likely will not until .NET Standard 2.0 early next year.

In the meantime, you won't be able to use SteamKit on .NET Core. We reference a lot of things that do not exist in Core, e.g. `Thread`, `SerializableAttribute`, `IsolatedStorage`, `Stream.Dispose()` and so on.

If you just want to use protobuf-net 2.1.0 in your own project on .NET Framework, rather add an Assembly Binding Redirect.

If you want to use SteamKit on .NET Core, you're going to be SOL anyway. I'm not sure what this pull request achieves.

**Comment by JustArchi** [comment]
> If you want to use SteamKit on .NET Core, you're going to be SOL anyway. I'm not sure what this pull request achieves.

It makes SK2 one step closer to .NET Core, and updates dependencies to newer version, which might bring various visible and invisible improvements. We should be asking why **not** update, instead of why should we.

**Comment by Netshroud** [comment]
@JustArchi I agree in principle. This PR is still good to have anyway... but I don't think it will have the desired effect.

**Comment by JustArchi** [comment]
Totally agreed with you there @Netshroud, SK2 is not that easy library to bring up with .NET Core dependencies, and it's definitely not a good idea to try make it work with .NET Core at all cost. Release of V2 should make things much easier to port, and this is probably when SK2 should slowly try to make a move 👍

**Comment by paralin** [comment]
@Netshroud whatever man. I realize it won't work with .Net standard now,
but keep your dependencies up to date!

No regens are required. Protogen has a newer version but I don't think it's
worth the effort to update it yet.

**Comment by Netshroud** [comment]
Is there any value in updating half a dependency?

**Comment by paralin** [comment]
????

You're updating a full dependency! protobuf-net.

**Comment by Netshroud** [comment]
Protogen being the other 'half'.

**Comment by Netshroud** [comment]
Turns out there is no newer version of protogen. 😅

**Comment by Netshroud** [comment]
Closing as superceded by #375.

**Pull request closed without merging**